query_engine/src/logical_optimizer/type_conversion.rs

```diff
@@ -281,16 +282,59 @@ impl<'a> ExprRewriter for TypeRewriter<'a> {
 }
 
 fn string_to_timestamp_ms(string: &str) -> Result<ScalarValue> {
+    // TODO(lee): remove following codes after PR(https://github.com/apache/arrow-rs/pull/3787) merged
```

Comment:
I suggest add a new function for this workaround, something like
```rust
fn string_to_timestamp_ms_workaround() -> Result<Option<_>>

// Then you can just 
let ts = string_to_timestamp_ms_workaround(str).unwrap_or_else(string_to_timestamp_nanos(string))
```

Reply:
ok，done